### PR TITLE
🔄 Support recursive includes in myst and tex

### PR DESCRIPTION
**Pull request by fwkoch**
This adds more robust support for "recursive includes" - these are where one file `include`s another file which `include`s a third file, etc.

Additionally:
- Circular `include` dependencies error nicely (including printing the "`include`-stack") - no infinite loops:
  image
- `tex` source files with `\input` macros resolve and reload recursively as well

Resolves:
- #548
- #797

### CODE REVIEW

File: packages/myst-cli/src/transforms/include.ts [hunk 1]
```diff
@@ -7,24 +7,64 @@ import type { VFile } from 'vfile';
...
+  };
+
+/**
+ * Return loadFile function
+ *
+ * Loaded file is added to original baseFile's dependencies.
+ */
+export const makeFileLoader = (session: ISession, baseFile: string) => (fullFile: string) => {
+  session.store.dispatch(
+    watch.actions.addLocalDependency({
```
Comment (fwkoch) [inline packages/myst-cli/src/transforms/include.ts:45]: These `include` dependencies _do not_ show up in the page json "dependency" field, but they are respected for live site reloading.

I think that's ok - the json "dependencies" are used for loading data from other pages; in the case of `include`, this data is actually moved to the page itself.

🤔 Maybe if we update `include` to allow including _partial_ snippets from md files we may need to load the rest of the data from that page...? (Similar to `embed` pulling in a single figure.) But for now, it's unnecessary.

**Comment by rowanc1** [comment]
Works really well locally in both latex and markdown - and a mix as well, which is kinda cool:

image

File: packages/myst-cli/src/transforms/include.ts (hunk 1 above)
Comment (rowanc1) [reply packages/myst-cli/src/transforms/include.ts:45]: Agree, the included files are a local convinience, not for how the final documents work.

**Comment by rowanc1** [comment]
Added some docs, this is good to go in my opinion!

image

**Review by rowanc1 — APPROVED** [review]
Code looks good, fun to see the implementation. Nice work @fwkoch.

**Pull request merged**